development/build/scripts.js

```diff
@@ -471,6 +471,33 @@ function createScriptTasks({
         version,
         applyLavaMoat,
       }),
+      () => {
+        // MV3 injects inpage into the tab's main world, but in MV2 we need
+        // to do it manually:
+        if (process.env.ENABLE_MV3) {
+          return;
+        }
+        // stringify inpage.js into itself, and then make it inject itself into the page
+        browserPlatforms.forEach((browser) => {
+          const inpagePath = path.join(
+            __dirname,
+            '../../',
+            `dist/${browser}/${inpage}.js`,
+          );
+          const textContent = JSON.stringify(readFileSync(inpagePath, 'utf8'))
+            .replace(/\u2028/gu, '\\u2028')
+            .replace(/\u2029/gu, '\\u2029');
+          const html = `{let d=document,s=d.createElement('script');s.textContent=${textContent};d.documentElement.appendChild(s).remove();}`;
+          writeFileSync(inpagePath, html, 'utf8');
+        });
+        // delete the inpage.js source map, as it no longer represents inpage.js
+        // and so `yarn source-map-explorer` can't handle it. It's also not
+        // useful anyway, as inpage.js is injected as a `script.textContent`,
+        // and not tracked in Sentry or browsers devtools anyway.
+        unlinkSync(
+          path.join(__dirname, `../../dist/sourcemaps/${inpage}.js.map`),
+        );
+      },
```

Comment:
I'm certain there is a better way to do the above.

If you didn't read the PR description (shame shame!): contentscript.js no longer depends on inpage.js being built fist, but I didn't update the gulp build process to decouple them.

I'd actually really love a suggestion/commit to do this in a more idiomatic gulp way.